test/src/Controller/SampleController.php

```diff
@@ -60,4 +64,50 @@ public function notSoBasicAction(Request $request)
 
         return $response;
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return JsonResponse|Response
+     */
+    public function indexAction(Request $request)
+    {
+        $accept = $request->headers->get('Accept');
+        if ('application/xml' === $accept) {
+            $content = '<?xml version="1.0" encoding="UTF-8"?><products><product><id>1</id><name>"Star-Wars T-shirt"</name><sku>"SWTS"</sku><price>5500</price></product><product><id>2</id><name>"Han Solo Mug"</name><sku>"HSM"</sku><price>5500</price></product></products>';
```

Comment:
Perhaps we could use http://symfony.com/doc/current/components/serializer.html for tests. And simply have some basic Product model and serialize it?